apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/components/CustomFilter.test.tsx

```
@@ -0,0 +1,202 @@
+    // and firstMountRef.current is not changed by the first useEffect.
+    const { rerender } = render(<CustomFilter survey={mockSurvey} />);
+    expect(mockUseResponseFilterState.resetState).not.toHaveBeenCalled();
+
+    const newSurvey = { ...mockSurvey, id: "survey-2" };
+    rerender(<CustomFilter survey={newSurvey} />);
+    expect(mockUseResponseFilterState.resetState).not.toHaveBeenCalled();
+  });
+
+  test("closes date picker when clicking outside", async () => {
+    const user = userEvent.setup();
+    let clickOutsideCallback: Function = () => {};
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)